facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java

```diff
@@ -162,6 +172,15 @@ public void prepareRepositoryMock() {
                     .findFirst()
                     .orElse(null);
         });
+        doAnswer(inv -> {
+            Integer id = inv.getArgument(0);
+            Object object = repositoryList.stream()
+                    .filter(a -> id.equals(a.getId()))
+                    .findFirst()
+                    .orElse(null);
+            repositoryList.remove(object);
+            return null;
+        }).when(repository()).delete(anyInt());
```

Comment:
Why do you use it in different order then the previous things are used?

Reply:
Delete returns void and there was a problem to use it inside when() method.

Reply:
right